pkg/controller/tailoredprofile/tailoredprofile_controller.go

```diff
@@ -152,6 +204,43 @@ func (r *ReconcileTailoredProfile) Reconcile(ctx context.Context, request reconc
 		}
 	}
 
+	ann := instance.GetAnnotations()
+	if _, ok := ann[cmpv1alpha1.DisableOutdatedReferenceValidation]; ok {
+		reqLogger.Info("Reference validation is disabled, skipping validation")
+	} else if isValidationRequired(instance) {
+		reqLogger.Info("Validating TailoredProfile")
+		pruneOutdated := false
+		if _, ok := ann[cmpv1alpha1.PruneOutdatedReferencesAnnotationKey]; ok {
+			pruneOutdated = true
+		}
+		// remove any deprecated variables or rules from the tailored profile
+		doContinue, variableNeedtoBeDeprecatedList, ruleNeedToBeDeprecatedList, err := r.handleDeprecation(instance, reqLogger, pruneOutdated)
+		if err != nil {
+			return reconcile.Result{}, err
+		}
+		if !doContinue {
+			return reconcile.Result{}, nil
+		}
+
+		doContinue, ruleNeedToBeMigratedList, err := r.handleRulePruning(instance, reqLogger, pruneOutdated)
+		if err != nil {
+			return reconcile.Result{}, err
+		}
+		if !doContinue {
+			return reconcile.Result{}, nil
+		}
+
+		warningMsg := generateWarningMessage(variableNeedtoBeDeprecatedList, ruleNeedToBeDeprecatedList, ruleNeedToBeMigratedList)
+		// check if warning message matches the previous warning message
+		// if it does, we don't need to update the tp, if not we need to update it with the new warning message
+		if warningMsg != instance.Status.Warnings {
```

Comment:
We might need to check if the warningMsg is an empty string, if so then we should make sure we remove the warning from the status.

Reply:
just had it tested, it turns out the `omitempty` will remove the field if it is an empty string.